IdeasTracker/IdeasTracker/View/LoginView.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  LoginView.swift
+//  IdeasTracker
+//
+//  Created by Wen Yu Ge on 2024-04-19.
+//
+
+import Foundation
+import SwiftUI
+
+struct LoginView: View {
+    @State private var email: String = ""
+    @State private var password: String = ""
+    @State private var isRegister: Bool = false
+    @FocusState private var focusedTextField: FormTextField?
+    @EnvironmentObject private var loginViewModel: LoginViewModel
+    @EnvironmentObject private var router: Router
+    
+    enum FormTextField {
+        case email, password
+    }
+    
+    private func handleRegister() async {
+        await loginViewModel.register(
+            email: email,
+            password: password
+        )
+    }
+    
+    private func handleLogin() async {
+        await loginViewModel.login(
+            email: email,
+            password: password)
+        router.pushReplacement(.ideas)
+    }
```

Comment:
If either of these calls fail, the user won't know. In the login case, they'll get pushed to the ideas screen even if login fails

Reply:
<img width="444" alt="Screenshot 2024-04-29 at 10 47 13 PM" src="https://github.com/appwrite/demos-for-apple/assets/29069505/1afcb469-d01c-4d63-adcd-72a55beccf8e">

Reply:
I'll add in a conditional bit into the form to show smthing simple. Ideally I'd toast but it's out of scope